qt_gui/src/qt_gui/main.py

```diff
@@ -193,18 +193,20 @@ def _add_reload_paths(self, reload_importer):
 
     def _check_icon_theme_compliance(self):
         from python_qt_binding.QtGui import QIcon
-        # TODO find a better way to verify Theme standard compliance
-        if QIcon.fromTheme('document-save').isNull() or \
-           QIcon.fromTheme('document-open').isNull() or \
-           QIcon.fromTheme('edit-cut').isNull() or \
-           QIcon.fromTheme('object-flip-horizontal').isNull():
-            if platform.system() == 'Darwin' and \
-                    '/usr/local/share/icons' not in QIcon.themeSearchPaths():
-                QIcon.setThemeSearchPaths(QIcon.themeSearchPaths() + ['/usr/local/share/icons'])
-            original_theme = QIcon.themeName()
-            QIcon.setThemeName('Tango')
-            if QIcon.fromTheme('document-save').isNull():
-                QIcon.setThemeName(original_theme)
+        try:
+            # if themeName is defined we are on Linux, otherwise try to find the package
+            # qt_gui_icons
+            if(QIcon.themeName() == ''):
+                icon_paths = QIcon.themeSearchPaths()
+                _, package_path = get_resource('packages', 'qt_gui_icons')
+                icon_paths.append(os.path.join(
+                    package_path, 'share', 'qt_gui_icons', 'resource', 'icons', 'Tango'))
+                QIcon.setThemeSearchPaths(icon_paths)
+                QIcon.setThemeName('scalable')
+        except LookupError:
```

Comment:
The `try` / `except` block should be as small as possible. Please reduce the scope to only contain the function call which is expected to raise.